### ISSUE TITLE
Revert back CSRF but improve documentation

### ISSUE THREAD

**Issue by mbhave** [spring-projects/spring-boot#12267]

**Comment by mbhave** [comment]
@lanoxx following up on your comment [here](https://github.com/spring-projects/spring-boot/issues/11170#issuecomment-368779644)

> please consider adding a big warning somewhere in the test

which test were you implying adding a warning to? I want to make sure I don't miss that when documenting this.

**Comment by lanoxx** [comment]
I was not accurate in my comment. Rather then `... somewhere in the test` it should read `... somewhere in the official documentation`. 

The reason I got caught by this was that I was running Unit tests which tested with an active Spring Security Context. As we are still in a prototype stage we have not yet added CSRF support to our frontend and we run unit tests without CSRF support. While I was using Spring Boot 1.5.9 CSRF was disabled by default, despite what the documentation said, and so this was never a problem for us. Then I changed to a manually configured Spring Security Configuration which caused the CSRF setting to become active and in turn this caused my Unit tests to fail.

**Regarding this issue:**
I would expect that somewhere in the official reference documentation a paragraph like this comes up:

> The default setting for CSRF has changed between 1.5.10 and 1.5.11. In 1.5.10 the setting was set to false and in 1.5.11 it was changed to true to match the default settings of Spring Security and to improve default security. Be aware that this is a breaking change. Upgrading users have two options. Either ensure that both your tests and clients implement CSRF support or consider to disable CSRF explicitly to retain the pre 1.5.11 behavior. If you disable CSRF you will not be protected from CSRF attacks.

Feel free to rephrase as you see fit.

Thank you for looking into this.
